python/tests/test_adjoint_jax.py

```diff
@@ -22,228 +22,251 @@
-    self.simulation.run(until=100)
-    monitor_values = mpa.utils.gather_monitor_values(self.monitors)
-    self.assertEqual(monitor_values.dtype, onp.complex128)
-    self.assertEqual(monitor_values.shape,
-                     (len(self.monitors), len(self.frequencies)))
-
-  def test_design_region_monitor_helpers(self):
-    design_region_monitors = mpa.utils.install_design_region_monitors(
-      self.simulation,
-      self.design_regions,
-      self.frequencies,
-    )
-    self.simulation.run(until=100)
-    design_region_fields = mpa.utils.gather_design_region_fields(
-      self.simulation,
-      design_region_monitors,
-      self.frequencies,
+    design_regions = [
+        mpa.DesignRegion(
+            mat_grid,
+            volume=mp.Volume(
+                center=mp.Vector3(),
+                size=mp.Vector3(
+                    design_region_shape[0],
+                    design_region_shape[1],
+                    0,
+                ),
+            ),
+        )
+    ]
+
+    geometry = [
+        mp.Block(center=mp.Vector3(x=-design_region_shape[0] / 2 -
+                                   wg_length / 2 - pml_width / 2),
+                 material=si,
+                 size=mp.Vector3(wg_length + pml_width, wg_width,
+                                 0)),  # left wg
+        mp.Block(center=mp.Vector3(x=+design_region_shape[0] / 2 +
+                                   wg_length / 2 + pml_width / 2),
+                 material=si,
+                 size=mp.Vector3(wg_length + pml_width, wg_width,
+                                 0)),  # right wg
+        mp.Block(center=design_regions[0].center,
+                 size=design_regions[0].size,
+                 material=mat_grid),  # design region
+    ]
+
+    simulation = mp.Simulation(
+        cell_size=mp.Vector3(sx, sy),
+        boundary_layers=[mp.PML(pml_width)],
+        geometry=geometry,
+        sources=sources,
+        resolution=sim_resolution,
     )
 
-    self.assertIsInstance(design_region_fields, list)
-    self.assertEqual(len(design_region_fields), len(self.design_regions))
-
-    self.assertIsInstance(design_region_fields[0], list)
-    self.assertEqual(len(design_region_fields[0]), len(mpa.utils._ADJOINT_FIELD_COMPONENTS))
+    monitor_centers = [
+        mp.Vector3(-sx / 2 + pml_width + source_to_pml + source_to_monitor),
+        mp.Vector3(sx / 2 - pml_width - source_to_pml - source_to_monitor),
+    ]
+    monitor_size = mp.Vector3(y=wg_width + 2 * wg_padding)
 
-    for value in design_region_fields[0]:
-      self.assertIsInstance(value, onp.ndarray)
-      self.assertEqual(value.ndim, 4)  # dims: freq, x, y, pad
-      self.assertEqual(value.dtype, onp.complex128)
+    monitors = [
+        mpa.EigenmodeCoefficient(simulation,
+                                 mp.Volume(center=center, size=monitor_size),
+                                 mode=1,
+                                 forward=True,
```

Comment:
Looks like this branch was not rebased from the master branch properly because the `forward` parameter on this line rather than being a constant is swept over `True` and `False`. Also, the line below is `decimation_factor=5` in the master branch.

Reply:
Good catch. Fixed!